applications/MeshingApplication/custom_utilities/mmg/mmg_utilities.cpp

```diff
@@ -3808,7 +3810,8 @@ void MmgUtilities<TMMGLibrary>::GenerateMeshDataFromModelPart(
         }
     }
 
-    r_nodes_array.Unique();
+    // This needs to be somewhere else.
+    // r_nodes_array.Unique();
```

Comment:
Someone from the meshing can tall me why we need to reorder and remove duplicates from the modelpart here? A workaround would be to create a print modelpart but we cannot modify the original in a write function

Reply:
The problem is the operation or where the operation is placed?

Reply:
Where it is placed. 

Right now is in a write method, which in IO needs a `const & ModelPart`, so we cannot do that. If the reorder is needed needs to be placed outside (which makes sense) but not sure where.

Reply:
The code you modified I don't think it will work. You cannot have const model part in the function that literally is retrieving the new mesh

Reply:
Yep that's my point, that function should not be in charge of doing that. Ideally you should have:

```c++
void foo() {
    ModelPart my_model = GenerateModel()
    Write(my_model)
}

but for some reason, in the meshing, the `GenerateModel()` part is done inside the write. I am trying to understand where this write code is being called to split it in two parts (the part that modifies the modelpart and the part that actually writes it)

Reply:
> including the "authority" stuff are merged

that's on me and will likely take ages